### ISSUE TITLE
cannot override sts endpoint

### ISSUE THREAD

**Issue by pr8kerl** [jtblin/kube2iam#164]
We have been using kube2iam for some time now with great success - so thank you. 

However we have had a few issues relating to the timely provisioning of credentials to pods. Recently in particular we had a situation where we were seeing a number of `InvalidTokenId` errors on pods with kube2iam annotations. These errors were occurring across a couple of kubernetes clusters and across two separate aws accounts. We raised a support ticket with AWS who said there was no sts service issue at that time, however they suggested we should be using our local region (ap-southeast-2) sts endpoint instead of the default endpoint (us-east-1). When using the default sts endpoint, any credentials created there need to be replicated to the region where they will be used. Our pods appeared to be attempting to use the sts credentials before they had replicated to our region - hence the `InvalidTokenId` error.

There have been a couple of other issues relating to the sts endpoint and region such as #72 and #103 

The answer to those issues was to set the AWS_REGION environment variable. However setting the region in this way does not appear to affect the STS endpoint used which always seems to default to `sts.amazonaws.com` regardless of the use of AWS_REGION or AWS_DEFAULT_REGION vars.

I've done some local tests with the codebase and found that the only way to ensure the endpoint is set correctly is to set it explicitly with the sts session Config and to also set the region.
This behaviour seems to be alluded to [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html#id_credentials_temp_enable-regions_writing_code)

Would you agree that this is the current behaviour?? That the use of region env vars does not affect the sts endpoint in use.

If so, would you be open to a PR to allow the use of a cli flag to set the sts endpoint? 

Thanks in advance.

**Comment by jrnt30** [comment]
I have not dug into the implementation details of the STS client specifically as closely as you have.  If you look at the CloudTrail logs that might shed some light on where, when the env vars are properly configured, the STS token is created at. 

If you're assessment is correct, that it is not respecting that Env flag, then a PR with additional configuration would seem reasonable and accepted, yes. 

Thanks for the diligence and detail!

**Comment by jtblin** [comment]
> This paragraph also describes the current

Which paragraph? :)

**Comment by pr8kerl** [comment]
whoops. I think I got sidetracked at that point. :)

I may have been about to link to this [paragraph](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html#sts-regionalization) which also dances around the fact that the default endpoint for every region is us-east-1. I haven't been able to find a documented statement saying that the endpoint is not affected by the AWS_REGION env vars. But by trial and error with the golang sdk endpoints package and the kube2iam code base, it does indeed appear to be the case that the only way to set the regional endpoint is by setting it explicitly in the code.

**Comment by SharpEdgeMarshall** [comment]
I was investigating this improvement some time ago and I thought to implement it, but never had time.

@jtblin From doc
> Writing Code to Use AWS STS Regions
After you activate a region for your AWS account, you can direct AWS STS API calls to that region. The following Java code snippet demonstrates how to configure an AWSSecurityTokenServiceClient object to make requests from the EU (Ireland) (eu-west-1) region with the setEndpoint method.
```
AWSSecurityTokenServiceClient stsClient = new AWSSecurityTokenServiceClient();
stsClient.setEndpoint("sts.eu-west-1.amazonaws.com");
```
> **Important
Do not use the setRegion method to set a regional endpoint for AWS STS because, for backward compatibility, that method continues to resolve to the original single global endpoint of sts.amazonaws.com.**

> In the example, the first line instantiates an AWSSecurityTokenServiceClient object called stsClient. The second line configures the stsClient object by calling its setEndpoint method and passing the URL of the endpoint as the only parameter. All API calls that use this stsClient object are now directed to the specified endpoint.

**Comment by jrnt30** [comment]
Fixed via #166
